.github/workflows/dev.yml

```diff
@@ -32,10 +32,7 @@ jobs:
         run: docker exec sporkfish-container pytest -sv
       - name: Check typing (mypy)
         run: docker exec sporkfish-container mypy --install-types --non-interactive
-      - name: Check code format (black)
-        run: docker exec sporkfish-container black --check .
-      - name: Check imports format (isort)
-        run: docker exec sporkfish-container isort --check --profile black .
-        
-     
-      
\ No newline at end of file
+      - name: Check code and imports format (ruff)
+        run: docker exec sporkfish-container ruff check .
```

Comment:
Sanity checking: Is my understanding correct that the reasons for replacing black/isort with ruff are (based on [this](https://www.packetcoders.io/whats-the-difference-black-vs-ruff/#:~:text=The%20choice%20between%20Black%20and,Ruff%20is%20a%20great%20choice.)):

- ruff is faster (not like black was slow tbf?)
- ruff does what black and isort would do (formatting), as well as [linting](https://code.visualstudio.com/docs/python/linting#:~:text=Linting%20is%20distinct%20from%20Formatting,only%20restructures%20how%20code%20appears.)

Reply:
The latter, plus I couldn't get auto isort to work